test/0.8.9/staking-router/staking-router-deposits.test.js

```diff
@@ -0,0 +1,256 @@
+        from: depositor,
+      })
+      const currentBlockNumber = await web3.eth.getBlockNumber()
+
+      assert.equals(await depositContract.totalCalls(), 100, 'invalid deposits count')
+
+      // on deposit we return balance to Lido
+      assert.equals(await web3.eth.getBalance(lido.address), ETH(32), 'invalid lido balance')
+      assert.equals(await web3.eth.getBalance(router.address), 0, 'invalid staking_router balance')
+
+      assert.equals(await lido.getBufferedEther(), ETH(32), 'invalid total buffer')
+
+      assert.emits(receipt, 'Unbuffered', { amount: ETH(depositsCount * 32) })
+
+      const lastModuleBlock = await router.getStakingModuleLastDepositBlock(curatedModuleId)
+      assert.equal(currentBlockNumber, +lastModuleBlock)
+    })
+  })
+
+  // the check for depositsValue == 0 occurs above in Lido.deposit(), so perhaps this check is redundant in SR.
+  // But if we untie the Lido contract from the StarkingRouter, we will need this check
+  describe('test deposit from staking router directly', async () => {
+    before(snapshot)
+    after(revert)
+
+    it('add module', async () => {
+      await router.addStakingModule(
+        'Curated',
+        operators.address,
+        10_000, // 100 % _targetShare
+        1_000, // 10 % _moduleFee
+        5_000, // 50 % _treasuryFee
+        { from: voting }
+      )
+      curatedModuleId = +(await router.getStakingModuleIds())[0]
+    })
+
+    it('prepare node operators', async () => {
+      // balance are initial
+      assert.equals(await web3.eth.getBalance(lido.address), ETH(1))
+      assert.equals(await web3.eth.getBalance(router.address), 0)
+
+      const sendEthForKeys = ETH(101 * 32 - 1)
+      const totalPooledEther = ETH(101 * 32)
+
+      await web3.eth.sendTransaction({ value: sendEthForKeys, to: lido.address, from: stranger })
+      assert.equals(await lido.getBufferedEther(), totalPooledEther)
+
+      // updated balance are lido 100 && sr 0
+      assert.equals(await web3.eth.getBalance(lido.address), totalPooledEther)
+      assert.equals(await web3.eth.getBalance(router.address), 0)
+
+      // prepare node operators
+      await operators.addNodeOperator('1', ADDRESS_1, { from: voting })
+      await operators.addNodeOperator('2', ADDRESS_2, { from: voting })
+
+      // add 150 keys to module
+      const keysAmount = 50
+      const keys1 = genKeys(keysAmount)
+      await operators.addSigningKeys(0, keysAmount, keys1.pubkeys, keys1.sigkeys, { from: voting })
+      await operators.addSigningKeys(0, keysAmount, keys1.pubkeys, keys1.sigkeys, { from: voting })
+      await operators.addSigningKeys(0, keysAmount, keys1.pubkeys, keys1.sigkeys, { from: voting })
+
+      await operators.setNodeOperatorStakingLimit(0, 100000, { from: voting })
+      await operators.setNodeOperatorStakingLimit(1, 100000, { from: voting })
+    })
+
+    it('zero deposits just updates module lastDepositBlock', async () => {
```

Comment:
I suggest having a similar test starting from the Lido contract.
Because we re-enabled this thing at the tooling team's request and prevent any future regressions.

Reply:
Added a similar test for the Lido + StakingRouter setup: https://github.com/lidofinance/lido-dao/pull/656/commits/5f4f315c9190db7f4775687ce1c337387a1718f4

Reply:
:rocket: ty!